procfile/procfile.go

```diff
@@ -47,11 +95,16 @@ func Parse(r io.Reader) (Procfile, error) {
 
 // ParseProcfile takes a byte slice representing a YAML Procfile and parses it
 // into a Procfile.
-func ParseProcfile(b []byte) (Procfile, error) {
-	p, err := parseStandardProcfile(b)
+func ParseProcfile(b []byte) (p Procfile, err error) {
+	p, err = parseStandardProcfile(b)
```

Comment:
does it make sense to have some sort of version specification in the Procfile?

I know vagrant did this with Vagrantfiles. I would think of `StandardProcfile` as v1, `ExtendedProcfile` as v2, you might have some v3, v4 etc in the future that you don't know about yet. This current method works for the two that are markedly different, but I think could get confusing in the future and implicitly being dropped into one format when you wanted to be in another.